### PR TITLE
Container nested components

### PR DESCRIPTION
**Pull request by charles474**
Fixes a bug where nested component(s) value(s) would not be set correctly within a container/ collections

### CODE REVIEW

**Review by LewisClark-HO — APPROVED** [review]
Looks good to me :)

File: src/utils/Component/elevateNestedComponents.test.js [hunk 1]
```diff
@@ -107,4 +107,45 @@ describe('utils.Component.elevateNestedComponents', () => {
     ]);
   });
 
+  it('should return the original array with elevated nested components when a component has a full_path', () => {
```
Comment (charles474) [inline src/utils/Component/elevateNestedComponents.test.js:110]: I'm not entirely sure about this test. Feels very much like a duplicate test.

File: src/utils/Component/elevateNestedComponents.js [hunk 2]
```diff
@@ -31,7 +33,8 @@ const elevateNestedComponents = (components, data) => {
     component.data?.options?.forEach((option) => {
       // If this option has nested components and is
       // selected, then add its nested components to the array.
-      if (Array.isArray(option.nested) && data[component.id] === option.value) {
+      const sourceData = data?.[component.id] || Data.getSource(data, component.full_path);
```
Comment (LewisClark-HO) [inline src/utils/Component/elevateNestedComponents.js:36]: Very minor, but don't need the `?` here after `data` as that check is already done in the containing `if`.

**Review by LewisClark-HO — APPROVED** [review]
LGTM

File: src/utils/Component/elevateNestedComponents.js (hunk 2 above)
Comment (LewisClark-HO) [reply src/utils/Component/elevateNestedComponents.js:36]: complete brain fart, please ignore

Comment (charles474) [reply src/utils/Component/elevateNestedComponents.js:36]: Thanking you 🙏

**Pull request merged**